tasks/request_status_for_job/test/unit_tests/test_request_status_for_job_unit.py

```diff
@@ -0,0 +1,213 @@
+                orca_status.value AS \"{request_status_for_job.OUTPUT_STATUS_KEY}\"
+            FROM
+                orca_recoveryjob
+            JOIN orca_status ON orca_recoveryjob.status_id=orca_status.id
+            WHERE
+                job_id = %s
+            """,
+                                                  db_connect_info, (job_id,))
+        mock_result_to_json.assert_called_once_with(mock_single_query.return_value)
+
+        self.assertEqual(mock_result_to_json.return_value, result)
+
+    @patch('database.single_query')
+    def test_get_granule_status_entries_for_job_error_wrapped_in_database_error(
+            self,
+            mock_single_query: MagicMock
+    ):
+        """
+        If error is raised when contacting DB, error should be raised as a DatabaseError.
+        """
+        job_id = uuid.uuid4().__str__()
+        db_connect_info = Mock()
+
+        mock_single_query.side_effect = database.DbError()
+
+        with self.assertRaises(request_status_for_job.DatabaseError):
+            request_status_for_job.get_granule_status_entries_for_job(job_id, db_connect_info)
+
+    @patch('database.single_query')
+    def test_get_status_totals_for_job_happy_path(
+            self,
+            mock_single_query: MagicMock
+    ):
+        """
+        Basic path with all information present.
+        """
+        job_id = uuid.uuid4().__str__()
+        db_connect_info = Mock()
+
+        mock_single_query.return_value = [{'value': 'status0', 'total': 5}, {'value': 'status1', 'total': 10}]
+
+        result = request_status_for_job.get_status_totals_for_job(job_id, db_connect_info)
+
+        mock_single_query.assert_called_once_with(f"""
+            with granule_status_count AS (
+                SELECT status_id
+                    , count(*) as total
+                FROM orca_recoveryjob
+                WHERE job_id = %s
+                GROUP BY status_id
+            )
+            SELECT value
+                , coalesce(total, 0) as total
+            FROM orca_status os
+            LEFT JOIN granule_status_count gsc ON (gsc.status_id = os.id)""",
+                                                  db_connect_info, (job_id,))
+
+        self.assertEqual({'status0': 5, 'status1': 10}, result)
+
+    @patch('database.single_query')
+    def test_get_status_totals_for_job_error_wrapped_in_database_error(
+            self,
+            mock_single_query: MagicMock
+    ):
+        """
+        If error is raised when contacting DB, error should be raised as a DatabaseError.
+        """
+        job_id = uuid.uuid4().__str__()
+        db_connect_info = Mock()
+
+        mock_single_query.side_effect = database.DbError()
+
+        with self.assertRaises(request_status_for_job.DatabaseError):
+            request_status_for_job.get_status_totals_for_job(job_id, db_connect_info)
+
+    def test_task_job_id_cannot_be_none(self):
```

Comment:
Not sure the purpose of this test here. If we are not passing an async_operation_id wouldn't we expect a specific exception return with HTTP status code?

Reply:
That happens in 'handler'. 'task' has a redundant check that raises an error. I could probably just remove the check at this point.